### ISSUE TITLE
Make Stationary Gas generators spawn in appropriate locations

### ISSUE THREAD

**Issue by JonathanLochridge** [CleverRaven/Cataclysm-DDA#55030]
### Is your feature request related to a problem? Please describe.

I saw that power grids were in. And then I remembered there are a lot of building types that tend to have backup generators IRL. Most notably hospitals and some factories/industrial facilities. Since Hospitals want to keep monitors running even if the grid goes down. And some factories are kept running 24/7 due to the startup time/cost to get it going again. Logically most of them should already be running or empty at game start. 

### Solution you would like.
1: Take the existing 2 tile gas generator and spawn it in appropriate locations:
-construction sites (hooked up to lights)(Sometimes they might be solar light rigs instead with a battery.
-light industry(sometimes)
-Hospital(always)
-any factory buildings(I think we have these? But maybe they were only implemented in a planned city expansion and never merged?

2: make new generator variant/vehicle with included lights.

### Describe alternatives you have considered.

-Add generators as a dedicated unique building tile like the planned building solar panels. see PR #54111

**Comment by Alex-Folts** [comment]
> ### Is your feature request related to a problem? Please describe.
> 1: add a 1-2 tile vehicle with an engine/alternator pre-connected to a building power grid.(Potentially some locations may have larger giant generators) 

Already exist in game, I never seen it spawned but 2-tile generator vehicle do exist, you can spawn it with debug menu and look.
Also there was this: https://github.com/CleverRaven/Cataclysm-DDA/pull/46632

**Comment by JonathanLochridge** [comment]
Huh, I didn't find it. Good to know well in that case the task changes to making it actually spawn in relevant places. Which is actually nicer for me since I am better with mapgen stuff than vehicles. So, I could probably add the existing generator to more locations. As well as check that is can connect to building power grids well.

The PR you linked never got merged. Since it says cloased rather than merged. It might be a good place to reference to borrow things from if whatever issues it has are fixed.

**Comment by coyo7e** [comment]
> > 1: add a 1-2 tile vehicle with an engine/alternator pre-connected to a building power grid.(Potentially some locations may have larger giant generators)
> 
> Already exist in game, I never seen it spawned but 2-tile generator vehicle do exist, you can spawn it with debug menu and look. Also there was this: #46632

I've found these in like, open sewer construction sites and stuff. And it's probably one of the things that taught me how to muss about with vehicles without being hugely intimidated by the system, after that.

Likely it's a spawn table thing where someone may need to pore through a LOT of building locations and tweak the generation tables slightly, but which is obscure enough that many folks don't even know where to begin, when they might otherwise be willing to put in the work and submit it.

**Comment by JonathanLochridge** [comment]
Huh, so only a particular type of construction site has it?

I can do some checking on it. But I wouldn't mid doing that poring as long as I can find a valid example somewhere. If not maybe I could try adding it in from scratch based on other places where spawning vehicles is implemented. 

I rarely go to the open sewer type construction site. I generally visit the other types. Mostly house construction sites.

**Comment by coyo7e** [comment]
> Huh, so only a particular type of construction site has it?
> 
> I can do some checking on it. But I wouldn't mid doing that poring as long as I can find a valid example somewhere. If not maybe I could try adding it in from scratch based on other places where spawning vehicles is implemented.
> 
> I rarely go to the open sewer type construction site. I generally visit the other types. Mostly house construction sites.

honestly I would do it myself if I could get through the huge pile of github docs when all I want is "how do I find the spawn tables for 'buildings etc' and adjust them?" But god bless you for being interested.

**Comment by JonathanLochridge** [comment]
Well, yeah the docs could use some work. They are decent though. I know where to find small item spawn tables. Since those are based in item groups. 
But I have like 5 other ideas that involve messing with vehicle spawns.
I know how to hardcode a specific vehicle spawn into a particular map. Or even a chance of spawning a particular vehicle. I think there are general tables. But, I am unsure where the vehicle equivalent to item groups are. Worst comes to worse I could manually add a spawn to the appropriate map files. It shouldn't be too many. Although, finding an appropriate vehicle group may save time for more of them.

Also, now that I think of it. Adding generators as something that can spawn in the bigger hardware stores could be interesting as well.

**Comment by coyo7e** [comment]
> Well, yeah the docs could use some work. They are decent though. I know where to find small item spawn tables. Since those are based in item groups. But I have like 5 other ideas that involve messing with vehicle spawns. I know how to hardcode a specific vehicle spawn into a particular map. Or even a chance of spawning a particular vehicle. I think there are general tables. But, I am unsure where the vehicle equivalent to item groups are. Worst comes to worse I could manually add a spawn to the appropriate map files. It shouldn't be too many. Although, finding an appropriate vehicle group may save time for more of them.
> 
> Also, now that I think of it. Adding generators as something that can spawn in the bigger hardware stores could be interesting as well.

if your would or could point me to which files need to be looked at, I do not need to waste tens of hours of time reading docs once I know where the "juice" is that needs a little kick.

**Comment by JonathanLochridge** [comment]
Eh most of the time it is only about 3-5 hours of doc reading per new thing I want to work on.
But, I will post a link when I find it. Sometimes I get more mileage by looking at whatever implemented maps/files are closest conceptually to what I am working on for reference. Although, there are a few guides for mapgen stuff that have some really good stuff that isn't apparent from the JSON.

I am currently in the process of learning markdown. But as I have been reading through them to work through stuff I have been gradually compiling a list of things to do to make it better.

Huh, I guess you had a bad experience? The guides for some stuff are pretty bad. Most projects I know with worse docs are simple enough not to need much in the way of docs.
Although, I have found the devs to be more willing to answer questions and help out than on other open source projects I have worked on/with. Particularly mlangsdorf. I tried working on the C++ a couple years ago and found it really difficult to get started. Mainly the compilation was the worst. But he helped me get my head around it then. Although the way a lot things is setup is complicated an confusing enough that I am not willing to dive back into the C++ unless I could set aside a couple weeks to really focus on it. And at this point my python skills have outstripped my C++ enough I would probably get 5x as much done with the same time if I worked on a python based project.
